### PR TITLE
adding productivity threshhold to amazon stonecutter_hut

### LINKED ISSUE #4580: No "No Rocks" warning from Amazon Stone Carver
Please fill out the relevant sections below and delete the rest.

**Describe the bug**
The Amazone Stone Carver does not send news when it is running out of stones.

**To Reproduce**
1. Open Attached Savegame
2. Check Stonecarver next to HQ.
3. Stonecarver ist at 0% productivity

**Expected behavior**

Stonecarver  sends news "Ages of stones is over" :-) or such.

**Version:**
 - OS: [macOS 11.1 / BigSur ]
 - Widelands Version: [1.0~git25105[bccc375@master]

[NoStoneWarn.wgf.zip](https://github.com/widelands/widelands/files/5761497/NoStoneWarn.wgf.zip)

### PR DESCRIPTION
**Pull request by hessenfarmer**
fixes #4580

fixed some issues from transifex as well

### CODE REVIEW

File: data/tribes/initialization/amazons/units.lua [hunk 1]
```diff
@@ -229,7 +229,7 @@ descriptions:new_tribe {
             helptexts = {
                purpose = {
                   -- TRANSLATORS: Helptext for an amazon ware: Ration, part 1
-                  pgettext("amazons_ware", "A small bite to keep miners strong and working. The scout also consumes rations on his scouting trips."),
+                  pgettext("amazons_ware", "A small bite to keep the workers digging for gold dust and granite strong and working. The scout also consumes rations on his scouting trips."),
```
Comment (klaus-halfmann) [inline data/tribes/initialization/amazons/units.lua]: If Amazones are female this should be: "..consumes rations on her scouting trips."

**Review by klaus-halfmann — COMMENTED** [review]
I dont want to start a gender discussion here but , well :-)

**Review by klaus-halfmann — CHANGES_REQUESTED** [review]
I am sorry. I get "No trees" messages but no "No Rocks".

(No Rocks is politically incorect - at least in German - anyway :-)

[NoStones.wgf.zip](https://github.com/widelands/widelands/files/5764129/NoStones.wgf.zip)

File: data/tribes/initialization/amazons/units.lua (hunk 1 above)
Comment (hessenfarmer) [reply data/tribes/initialization/amazons/units.lua]: ooops. You are right

**Comment by hessenfarmer** [comment]
did some testing. The message is triggered if 0 % is reached. The reason is that the stonecutters have a looped working programs with main program calling "cut_granite" like in mines. This can be easily merged into one program now as the reason for this is gone with the new productivity calculation. However I need to check the mines whether it is the same for them as well

**Comment by hessenfarmer** [comment]
mines work different and all other quarries already had been changed to a single work program

**Review by klaus-halfmann — APPROVED** [review]
OK for me now, although i takes quite some time for the stonecutters to complain, well.

Played some hour with Amazones now, still must get use to that tribe :-)

**Pull request merged**